src/main/kotlin/org/jitsi/jicofo/bridge/ExternalBridgeSelectionStrategy.kt

```diff
@@ -0,0 +1,110 @@
+package org.jitsi.jicofo.bridge
+
+import org.jitsi.utils.logging2.Logger
+import org.jitsi.utils.logging2.LoggerImpl
+import org.json.simple.JSONObject
+import org.json.simple.JSONValue
+import java.net.URI
+import java.net.http.HttpClient
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
+
+@Suppress("unused")
+class ExternalBridgeSelectionStrategy() : BridgeSelectionStrategy() {
+    private val httpClient: HttpClient = HttpClient
+        .newBuilder()
+        .connectTimeout(ExternalBridgeSelectionStrategyConfig.config.timeout)
+        .build()
+
+    private val logger: Logger = LoggerImpl(ExternalBridgeSelectionStrategy::class.simpleName)
+
+    private val fallbackStrategy: BridgeSelectionStrategy? by lazy {
+        val fallbackStrategyName = ExternalBridgeSelectionStrategyConfig.config.fallbackStrategy ?: return@lazy null
+        try {
+            val clazz = Class.forName("${javaClass.getPackage().name}.$fallbackStrategyName")
+            clazz.getConstructor().newInstance() as BridgeSelectionStrategy
+        } catch (e: Exception) {
+            val clazz = Class.forName(fallbackStrategyName)
+            clazz.getConstructor().newInstance() as BridgeSelectionStrategy
+        }
+    }
+
+    private fun fallback(
+        bridges: MutableList<Bridge>?,
+        conferenceBridges: MutableMap<Bridge, Int>?,
+        participantRegion: String?
+    ): Bridge {
+        if (fallbackStrategy == null) {
+            throw Exception("External bridge selection failed and no fallbackStrategy was provided.")
+        }
+        return fallbackStrategy!!.doSelect(bridges, conferenceBridges, participantRegion)
+    }
+
+    override fun doSelect(
+        bridges: MutableList<Bridge>?,
+        conferenceBridges: MutableMap<Bridge, Int>?,
+        participantRegion: String?
+    ): Bridge {
+        val url = ExternalBridgeSelectionStrategyConfig.config.url
+            ?: throw Exception("ExternalBridgeSelectionStrategy requires url to be provided")
+
+        val requestBody = JSONObject()
+        requestBody["bridges"] = bridges?.map {
```

Comment:
I just realized that you are providing the full list of bridges here. Can you make that optional (behind a config flag) please? We intend to use it with a service which already has the list of available bridges.

Reply:
It's necessary as long as the index is used in the response to identify the bridge. But if the JID is used as suggested, this can even be removed entirely rather than made optional IMO (our service also knows the list of available bridges).

But I was concerned how Jicofo would react if told to use a bridge it doesn't already know about. Even if not deliberate, I can imagine when bridges start up or shut down there could be brief periods where the external selection service has a different idea of available bridges than Jicofo.